frontend/packages/ceph-storage-plugin/src/components/modals/advanced-kms-modal/advanced-kms-modal.tsx

```diff
@@ -0,0 +1,272 @@
+  );
+  const [clientKey, setClientKey] = React.useState(kms.clientCert?.stringData['tls.key'] || '');
+  const [clientKeyFile, setClientKeyFile] = React.useState(kms.clientKeyFile || '');
+  const [providerNS, setProvideNS] = React.useState(kms.providerNamespace || '');
+  const [error, setError] = React.useState('');
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const caSecret: SecretKind = {
+      apiVersion: SecretModel.apiVersion,
+      kind: SecretModel.kind,
+      metadata: {
+        name: 'ocs-kms-ca',
+        namespace: CEPH_STORAGE_NAMESPACE,
+      },
+      stringData: {
+        'ca.cert': caCertificate,
+      },
+    };
+
+    const clientCertSecret: SecretKind = {
+      apiVersion: SecretModel.apiVersion,
+      kind: SecretModel.kind,
+      metadata: {
+        name: 'ocs-kms-client-cert',
+        namespace: CEPH_STORAGE_NAMESPACE,
+      },
+      stringData: {
+        'tls.cert': clientCertificate,
+      },
+    };
+
+    const clientKeySecret: SecretKind = {
+      apiVersion: SecretModel.apiVersion,
+      kind: SecretModel.kind,
+      metadata: {
+        name: 'ocs-kms-client-key',
+        namespace: CEPH_STORAGE_NAMESPACE,
+      },
+      stringData: {
+        'tls.key': clientKey,
+      },
+    };
+
+    const kmsAdvanced = {
+      ...kms,
+      backend: backendPath,
+      tls: tlsName,
+      providerNamespace: providerNS,
+      caCertFile: caCertificateFile,
+      clientCertFile: clientCertificateFile,
+      clientKeyFile,
+    };
+
+    caCertificate && caCertificate !== ''
+      ? (kmsAdvanced.caCert = caSecret)
+      : (kmsAdvanced.caCert = null);
+    clientCertificate && clientCertificate !== ''
+      ? (kmsAdvanced.clientCert = clientCertSecret)
+      : (kmsAdvanced.clientCert = null);
+    clientKey && clientCertificate !== ''
+      ? (kmsAdvanced.clientKey = clientKeySecret)
+      : (kmsAdvanced.clientKey = null);
+
+    setEncryptionDispatch(ActionType.SET_KMS_ENCRYPTION, mode, dispatch, kmsAdvanced);
+    close();
+  };
+  const readFile = (file: File, fn: Function, fileFn: Function) => {
+    const reader = new FileReader();
+    reader.onload = () => {
+      const input = reader.result;
+      fn(input.toString());
+    };
+    if (file) {
+      reader.readAsText(file, 'UTF-8');
+    } else {
+      fn('');
+      fileFn('');
+    }
+  };
+
+  const updateCaCert = (value: File, filename: string) => {
+    readFile(value, setCACertificate, setCACertificateFile);
+    setCACertificateFile(filename);
+  };
+
+  const updateClientCert = (value: File, filename: string) => {
+    readFile(value, setClientCertificate, setClientCertificateFile);
+    setClientCertificateFile(filename);
+  };
+
+  const updateClientKey = (value: File, filename: string) => {
+    readFile(value, setClientKey, setClientKeyFile);
+    setClientKeyFile(filename);
+  };
+
+  return (
+    <Form onSubmit={submit} key="pool-form-modal">
```

Comment:
nit: pool-form-modal
```suggestion
    <Form onSubmit={submit} key="">
```